ogr/ogrsf_frmts/openfilegdb/ogropenfilegdbdatasource_write.cpp

```diff
@@ -1669,6 +1670,191 @@ const GDALRelationship* OGROpenFileGDBDataSource::GetRelationship(const std::str
     return it->second.get();
 }
 
+
+/************************************************************************/
+/*                          AddRelationship()                           */
+/************************************************************************/
+
+bool OGROpenFileGDBDataSource::AddRelationship(std::unique_ptr<GDALRelationship>&& relationship,
+                                               std::string& failureReason)
+{
+    const auto relationshipName = relationship->GetName();
```

Comment:
I guess there must be restrictions on the validity of relationship names in FileGeodatabase. Perhaps similar to those of tables ? If so, I'm not sure if we should error out, or do name "laundering" as we do in ICreateLayer() / CreateField()

Reply:
Good point! Was there any particular process you used to test layer name validity?

Reply:
> Was there any particular process you used to test layer name validity?

I just it did from Esri doc available on the web.
As the name is used to format a path, I guess at least anti-slash cannot be used, but probably much more.
Otherwise have some trial&error tests against Esri softare